app/src/main/java/org/schabi/newpipe/ui/components/video/comment/CommentSection.kt

```diff
@@ -0,0 +1,179 @@
+    val state = rememberLazyListState()
+
+    Surface(color = MaterialTheme.colorScheme.background) {
+        LazyColumnThemedScrollbar(state = state) {
+            LazyColumn(
+                modifier = Modifier.nestedScroll(nestedScrollInterop),
+                state = state
+            ) {
+                when (uiState) {
+                    is Resource.Loading -> {
+                        item {
+                            LoadingIndicator(modifier = Modifier.padding(top = 8.dp))
+                        }
+                    }
+
+                    is Resource.Success -> {
+                        val commentInfo = uiState.data
+                        val count = commentInfo.commentCount
+
+                        if (commentInfo.isCommentsDisabled) {
+                            item {
+                                NoItemsMessage(R.string.comments_are_disabled)
+                            }
+                        } else if (count == 0) {
+                            item {
+                                NoItemsMessage(R.string.no_comments)
+                            }
+                        } else {
+                            // do not show anything if the comment count is unknown
+                            if (count >= 0) {
+                                item {
+                                    Text(
+                                        modifier = Modifier
+                                            .padding(start = 12.dp, end = 12.dp, bottom = 4.dp),
+                                        text = pluralStringResource(R.plurals.comments, count, count),
+                                        maxLines = 1,
+                                        style = MaterialTheme.typography.titleMedium
+                                    )
+                                }
+                            }
+
+                            when (comments.loadState.refresh) {
+                                is LoadState.Loading -> {
+                                    item {
+                                        LoadingIndicator(modifier = Modifier.padding(top = 8.dp))
+                                    }
+                                }
+
+                                is LoadState.Error -> {
+                                    item {
+                                        NoItemsMessage(R.string.error_unable_to_load_comments)
+                                    }
+                                }
+
+                                else -> {
+                                    items(comments.itemCount) {
+                                        Comment(comment = comments[it]!!) {}
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    is Resource.Error -> {
+                        item {
+                            NoItemsMessage(R.string.error_unable_to_load_comments)
```

Comment:
An error panel should be created to handle all of these cases and allow retrying, reporting and opening in browser.